docs/core/deploying/native-aot/diagnostics.md

```diff
@@ -0,0 +1,54 @@
+- NativeAOT application is neither a typical managed application (for example, no JIT) nor a typical native application (for example, has full GC support). The intent is to meet the reasonable diagnostics expectation of a product that is coming out of the .NET family and also the experience should feel familiar to someone who does production diagnostics on a C++ application.
+
+## Current NativeAOT diagnostic support
+
+Not all diagnostic features in NativeAOT are currently at parity with .NET. The following table summarizes NativeAOT diagnostic feature parity with .NET:
+
+| Feature | Fully Supported | Partially Supported | Not Supported |
+| - | - | - | - |
+| Build (Inner dev loop) diagnostics | <span aria-hidden="true">✔️</span><span class="visually-hidden">Fully supported</span> | | |
+| Observability | | <span aria-hidden="true">✔️</span><span class="visually-hidden">Partially supported</span> | |
+| Profiling | | <span aria-hidden="true">✔️</span><span class="visually-hidden">Partially supported</span> | |
+| Production debugging | | <span aria-hidden="true">✔️</span><span class="visually-hidden">Partially supported</span> | |
+| Heap analysis | | | <span aria-hidden="true">❌</span><span class="visually-hidden">Not supported</span> |
+
+### Build time diagnostics
+
+Build time (inner-dev-loop experience) diagnostic experience for a NativeAOT application will match exactly the .NET application experience since the application in build-time will run on .NET. Given the challenges in acquiring diagnostics in production environment and the current gap in diagnostic tools in NativeAOT, spending time in build time diagnostic in NativeAOT will be the best solution to build a performant and reliable NativeAOT application.
+
+### Observability support
+
+NativeAOT runtime supports [EventPipe](../../diagnostics/eventpipe.md) which will allow NativeAOT applications to easily trace their applications. EventPipe support will also allow most .NET diagnostic tools like [dotnet-trace](../../diagnostics/dotnet-trace.md), [dotnet-counter](../../diagnostics/dotnet-counters.md), and [dotnet-monitor](../../diagnostics/dotnet-monitor.md) to work seamlessly with NativeAOT applications.
+
+NativeAOT also provides partial support for [OpenTelemetry](../../diagnostics/observability-with-otel.md) and some [well-known event providers](../../diagnostics/well-known-event-providers.md). Not all [runtime events](../../../fundamentals/diagnostics/runtime-events.md) are supported in NativeAOT.
+
+### Profiling support
+
+"Microsoft-DotNETCore-SampleProfiler" provider is not currently supported in NativeAOT. Platform specific tools like [PerfView](https://github.com/microsoft/perfview), and [Perf](https://perf.wiki.kernel.org/index.php/Main_Page) can be used to collect CPU samples of a NativeAOT application.
+
+### Production debugging
```

Comment:
This seems more about debugging in general?

Reply:
Added the part about observability as @jkotas suggested. Debugging experience is drastically difference in build and production phases for native AOT and wanted to differentiate in the doc.